src/pyclass_init.rs

```diff
@@ -1,9 +1,60 @@
 //! Contains initialization utilities for `#[pyclass]`.
+//!
+//! # Background
```

Comment:
although it seems fairly straightforward once compiled like this, it took quite a bit of experimentation to make sure I was getting things correct so I wanted to document my findings somewhere